consensus/hotstuff/voteaggregator/interfaces.go

```diff
@@ -0,0 +1,24 @@
+package voteaggregator
+
+import (
+	"github.com/onflow/flow-go/consensus/hotstuff"
+	"github.com/onflow/flow-go/consensus/hotstuff/model"
+	"github.com/onflow/flow-go/model/flow"
+)
+
+// LazyInitCollector is a helper structure that is used to return collector which is lazy initialized
+type LazyInitCollector struct {
+	Collector hotstuff.VoteCollector
+	Created   bool // whether collector was created or retrieved from cache
+}
+
+type VoteCollectors interface {
+	// GetOrCreateCollector performs lazy initialization of hotstuff.VoteCollector
+	// collector is indexed by blockID and view
+	GetOrCreateCollector(view uint64, blockID flow.Identifier) (LazyInitCollector, error)
```

Comment:
Why returning a concrete type instead of a `VoteCollector` interface?

Why the caller (VoteAggregator) needs to know the vote collector is lazily inited or not?

Reply:
We can return concrete type. In approvals collecting it was beneficial to know when it was created vs accessed.